lib/core/include/scipp/core/value_and_variance.h

```diff
@@ -71,8 +71,10 @@ template <class B, class E>
 constexpr auto pow(const ValueAndVariance<B> base, const E exponent) noexcept {
   const auto pow_1 = numeric::pow(base.value, exponent - 1);
   const auto var_factor = std::abs(exponent) * pow_1;
-  return ValueAndVariance{pow_1 * base.value,
-                          var_factor * var_factor * base.variance};
+  const auto pow_0 = (base.value == 0 && exponent >= 0 && exponent < 1)
+                         ? 0
```

Comment:
- Shouldn't 0**0 give 1?
- Why do we need the `exponent < 1`, or the sign check? Isn't it just zero everywhere if the base is zero (except of the exponent also is)?

Reply:
1. Good catch.
2. Yeah that's probably right, it's an unnecessary check.